### PR TITLE
POW Registration allowance/disallowance

### PR DESCRIPTION
**Pull request by Rubberbandits**
This PR adds a sudo call which separately manages whether or not POW registration is enabled for a subnet.

### CODE REVIEW

File: pallets/subtensor/src/registration.rs [hunk 1]
```diff
@@ -317,7 +317,7 @@ impl<T: Config> Pallet<T> {
 
         // --- 3. Ensure the passed network allows registrations.
         ensure!(
-            Self::if_subnet_allows_registration(netuid),
+            Self::get_network_pow_registration_allowed(netuid),
```
Comment (eduardogr) [inline pallets/subtensor/src/registration.rs:320]: is there some idiomatic reason for this name change?
I thjink the previous one has a good naming, maybe something like.... `Self::subnet_allows_registrations(netuid)` which could fit with the context and its semantics. What do you think?

Comment (Rubberbandits) [reply pallets/subtensor/src/registration.rs:320]: if_subnet_allows_registration was a duplicate of another function which was already implemented.

**Review by ifrit98 — APPROVED** [review]
LGTM! Only question is on some failing tests. Assume these panics are from flaky tests or not updated?

image

**Comment by Rubberbandits** [comment]
> LGTM! Only question is on some failing tests. Assume these panics are from flaky tests or not updated?
> 
> image

Yep, i do need to update these! Good catch

**Comment by Rubberbandits** [comment]
Burn adjustment tests fixed

**Review by ifrit98 — APPROVED** [review]
LGTM! 

Only issue is the `test_registration_difficulty_adjustment` fails.

**Comment by Rubberbandits** [comment]
> LGTM!
> 
> Only issue is the `test_registration_difficulty_adjustment` fails.

Fixed! Thank you for catching this.

**Review by shibshib — APPROVED** [review]

**Pull request merged**